source/robot_model/include/robot_model/Model.hpp

```diff
@@ -98,6 +110,34 @@ class Model {
    */
   unsigned int get_frame_id(const std::string& frame);
 
+  /**
+   * @brief Function that extracts package names from URDF string
+   * @param urdf string containing the URDF description of the robot
+   * @return set of package names
+  */
+  std::set<std::string> extract_package_name_from_urdf();
+
+  /**
+   * @brief Replaces the package name by the full path in the urdf file
+   * @param urdf string containing the URDF description of the robot
+   * @param package_name the name of the package to replace
+   * @param full_path the full path to replace the package name
+  */
+  void replace_package_by_full_path(const std::string& package_name, const std::string& full_path);
+
+  /**
+    * @brief Resolves the package paths in the URDF string and returns the resolved URDF string
+    * @param urdf string containing the URDF description of the robot
+  **/
+  void resolve_package_paths();
```

Comment:
You should be more explicit that here "resolve" actually means "find and replace all package paths with absolute paths using the meshloader callback"

You also write that it `returns the resolved URDF string` but it doesn't actually. As I commented elsewhere, I think this function should actually take a reference to the urdf string or stream and modify it by reference. It should also return the list of package paths.